src/sentry/integrations/slack/utils/users.py

```diff
@@ -74,3 +79,63 @@ def get_slack_data_by_user(
                 slack_data_by_user[user] = slack_info_by_email[email]
                 break
     return slack_data_by_user
+
+
+# USING SDK CLIENT
+
+
+@dataclass
+class SlackUserData:
+    email: str
+    team_id: str
+    slack_id: str
+
+
+def format_slack_data_by_user(
+    emails_by_user: Mapping[User, Sequence[str]], slack_info_by_email: dict[str, SlackUserData]
+) -> Mapping[User, SlackUserData]:
+    slack_data_by_user: MutableMapping[User, SlackUserData] = {}
+    for user, emails in emails_by_user.items():
+        # get overlap between user emails and emails in slack
+        user_slack_emails = set(emails) & set(slack_info_by_email.keys())
+        if user_slack_emails:
+            slack_data_by_user[user] = slack_info_by_email[list(user_slack_emails)[0]]
+    return slack_data_by_user
+
+
+def format_slack_info_by_email(users: dict[str, Any]) -> dict[str, SlackUserData]:
+    return {
+        member["profile"]["email"]: SlackUserData(
+            email=member["profile"]["email"], team_id=member["team_id"], slack_id=member["id"]
+        )
+        for member in users
+        if not member["deleted"] and member["profile"].get("email")
+    }
+
+
+def get_slack_data_by_user_via_sdk(
+    integration: Integration | RpcIntegration,
+    organization: Organization | RpcOrganization,
+    emails_by_user: Mapping[User, Sequence[str]],
+) -> Generator[Mapping[User, Mapping[str, str]], None, None] | None:
+    sdk_client = SlackSdkClient(integration_id=integration.id)
+    try:
+        users_list = sdk_client.users_list(limit=SLACK_GET_USERS_PAGE_SIZE)
+
+        for page in users_list:
+            users: dict[str, Any] = page.get("members")
+            slack_info_by_email = format_slack_info_by_email(users)
+            slack_data_by_user = format_slack_data_by_user(emails_by_user, slack_info_by_email)
+
+            yield slack_data_by_user
+    except SlackApiError as e:
```

Comment:
Can we narrow the scope of the `try` block to just the `sdk_client.users_list` statement?

Reply:
when i'm iterating through the results of `sdk_client.users_list` it will paginate through the API (the `SlackResponse` object has a `__next__` function that does this). i think it's possible that `SlackApiError` is raised while iterating through the pages, am I interpreting that correctly?

https://github.com/slackapi/python-slack-sdk/blob/v3.27.1/slack_sdk/web/slack_response.py#L118